### ISSUE TITLE
Massive swing in model performance with different Seed value

### ISSUE THREAD

**Issue by Anshul12256** [ultralytics/ultralytics#22497]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

I have trained two yolo models with identical parameters but with different seed values. For 1st model, I used a seed value of **0**; for the other, I used a seed value of **44**.

My training dataset consists of 65,500 training images and 3,000 validation images.

However, the Recall and Precision performance of both models differ significantly on the two test datasets that I have.

**Test Dataset 1**:
Model with seed **0** - TP: 56744  FN: 86250  FP: 549    Precision: 99.04    Recall: 39.68     F1 Score: 56.66 
Model with seed **44** - TP: 55682  FN: 87312  FP: 1202   Precision: 97.89    Recall: 38.94     F1 Score: 55.72

**Test Dataset 2**:
Model with seed **0** - TP: 3838   FN: 836    FP: 18     Precision: 99.53    Recall: 82.11     F1 Score: 89.99
Model with seed **44** - TP: 3701   FN: 973    FP: 29     Precision: 99.22    Recall: 79.18     F1 Score: 88.08
  
Can you please explain why there is this massive swing in the model's Recall and Precision Performance?

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Anshul12256, thank you for your interest in Ultralytics 🚀! This is an automated response — an Ultralytics engineer will also assist you here soon.

We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.

If this is a custom training ❓ Question, please provide as much information as possible, including:
-Exact ultralytics version and environment details (OS, Python, PyTorch, CUDA)
-Full training/eval commands and key args (model, imgsz, epochs, batch, seed, augmentations)
-Training logs and validation metrics
-Validation/inference scripts used to compute your Precision/Recall/F1
-A few representative dataset images and labels (or label stats) 📊
Also verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a Python>=3.8 environment with PyTorch>=1.8 to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including CUDA/CUDNN, Python and PyTorch preinstalled):

- Notebooks with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- Google Cloud Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- Amazon Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- Docker Image. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by Y-T-G** [comment]
The difference in scores is <1% for the larger test dataset and slightly larger for the smaller dataset which is normal because the smaller the test dataset, the more unstable the metrics.

That doesn't seem massive. How many epochs have you trained for?

**Comment by Anshul12256** [comment]
@Y-T-G Thanks for your reply. I have trained the model for 450 epochs and I have only trained for 1 class. 
The reason I call it massive is because 3% Recall difference is still large when my task is safety critical. For the same reason 1% Precision loss is also significant.

**Comment by Anshul12256** [comment]
@Y-T-G is there a way we can fix initialization on the weights and random initialization for augmentation?

Currently, I am training models with yolov8s.yaml, because I believe the seed factor affects both weights and data augmentation.

**Comment by Y-T-G** [comment]
You can save the model with a particular initialization using `model.save()` and then use that `.pt` for training

**Comment by Anshul12256** [comment]
@Y-T-G. Thanks for the suggestion. It worked well and now performance is comparable even with different seed.
